crates/nu-engine/src/env.rs

```diff
@@ -3,20 +3,59 @@ use nu_path::canonicalize_with;
 use nu_protocol::{
     ast::Expr,
     engine::{Call, EngineState, Stack, StateWorkingSet},
-    ShellError, Span, Value, VarId,
+    ShellError, Span, Type, Value, VarId,
 };
 use std::{
     collections::HashMap,
     path::{Path, PathBuf},
     sync::Arc,
 };
 
-const ENV_CONVERSIONS: &str = "ENV_CONVERSIONS";
+pub const ENV_CONVERSIONS: &str = "ENV_CONVERSIONS";
 
-enum ConversionResult {
-    Ok(Value),
-    ConversionError(ShellError), // Failure during the conversion itself
-    CellPathError, // Error looking up the ENV_VAR.to_/from_string fields in $env.ENV_CONVERSIONS
+enum ConversionError {
+    ShellError(ShellError),
+    CellPathError,
+}
+
+impl From<ShellError> for ConversionError {
+    fn from(value: ShellError) -> Self {
+        Self::ShellError(value)
+    }
+}
+
+/// Translate environment variables from Strings to Values.
+pub fn convert_env_vars(
+    stack: &mut Stack,
+    engine_state: &EngineState,
+    conversions: &Value,
+) -> Result<(), ShellError> {
+    let conversions = conversions.as_record()?;
+    for (key, conversion) in conversions.into_iter() {
+        if let Some(val) = stack.get_env_var_insensitive(engine_state, key) {
```

Comment:
I'm not objecting to this but I'm wondering if all these should be insensitive. I know `Path` should but I'm not sure if there's any harm having all of them case insensitive. Thoughts?

Reply:
I chose to make it insensitive to make it consistent with how `Instruction::LoadEnv` loads environment variables. https://github.com/nushell/nushell/blob/baf86dfb0ed57aefb75962ea6cfc1552cb44c2c4/crates/nu-engine/src/eval_ir.rs#L357

Reply:
People complaining about http_proxy and HTTP_PROXY is really why I was asking. Leaving it this way at least is not a breaking change. Thanks.